discovery_engine/lib/src/domain/models/document.dart

```diff
@@ -52,6 +53,15 @@ class Document {
     this.isActive = true,
   })  : documentId = DocumentId(),
         timestamp = DateTime.now().toUtc();
+
+  api.Document toApiDocument() => api.Document(
```

Comment:
it's not a concern for this PR, but maybe we could think about renaming one of the `Document` classes to avoid further confusion? I was thinking that the `Document` exposed to the app could be named `DocumentDto`, which is closer to the truth, but it would be a nicer api if we kept it and renamed the domain `Document` instead, maybe to `DocumentEntity`? wdyt? cc @acrrd

Reply:
I think either of those suggestions would be a little less confusing, yes.